pkg/runtime/bulk_subscriber.go

```diff
@@ -0,0 +1,507 @@
+					(*bulkResponses)[entryID].EntryID = response.EntryID
+					(*bulkResponses)[entryID].Error = errors.Errorf("RETRY required while processing bulk subscribe event for entry id: %v", response.EntryID)
+					hasAnyError = true
+				case pubsub.Success:
+					diag.DefaultComponentMonitoring.BulkPubsubIngressEvent(ctx, msg.pubsub, strings.ToLower(string(pubsub.Success)), msg.topic, elapsed)
+					(*bulkResponses)[entryID].EntryID = response.EntryID
+					(*bulkResponses)[entryID].Error = nil
+				case pubsub.Drop:
+					diag.DefaultComponentMonitoring.BulkPubsubIngressEvent(ctx, msg.pubsub, strings.ToLower(string(pubsub.Drop)), msg.topic, elapsed)
+					log.Warnf("DROP status returned from app while processing pub/sub event %v", response.EntryID)
+					(*bulkResponses)[entryID].EntryID = response.EntryID
+					(*bulkResponses)[entryID].Error = nil
+				default:
+					// Consider unknown status field as error and retry
+					diag.DefaultComponentMonitoring.BulkPubsubIngressEvent(ctx, msg.pubsub, strings.ToLower(string(pubsub.Retry)), msg.topic, elapsed)
+					(*bulkResponses)[entryID].EntryID = response.EntryID
+					(*bulkResponses)[entryID].Error = errors.Errorf("unknown status returned from app while processing bulk subscribe event %v: %v", response.EntryID, response.Status)
+					hasAnyError = true
+				}
+			} else {
+				log.Warnf("Invalid entry id received from app while processing pub/sub event %v", response.EntryID)
+				continue
+			}
+		}
+		for _, item := range msg.entries {
+			ind := entryIDIndexMap[item.EntryID]
+			if (*bulkResponses)[ind].EntryID == "" {
+				(*bulkResponses)[ind].EntryID = item.EntryID
+				(*bulkResponses)[ind].Error = errors.Errorf("Response not received, RETRY required while processing bulk subscribe event for entry id: %v", item.EntryID)
+			}
+		}
+		if hasAnyError {
+			return errors.New("Few message(s) have failed during bulk subscribe operation")
+		} else {
+			return nil
+		}
+	}
+
+	if statusCode == nethttp.StatusNotFound {
+		// These are errors that are not retriable, for now it is just 404 but more status codes can be added.
+		// When adding/removing an error here, check if that is also applicable to GRPC since there is a mapping between HTTP and GRPC errors:
+		// https://cloud.google.com/apis/design/errors#handling_errors
+		log.Errorf("non-retriable error returned from app while processing bulk pub/sub event: %s. status code returned: %v", body, statusCode)
+		diag.DefaultComponentMonitoring.BulkPubsubIngressEvent(ctx, msg.pubsub, strings.ToLower(string(pubsub.Drop)), msg.topic, elapsed)
+		populateBulkSubscribeResponsesWithError(msg.entries, bulkResponses, &entryIDIndexMap, nil)
+		return nil
+	}
+
+	// Every error from now on is a retriable error.
+	log.Warnf("retriable error returned from app while processing bulk pub/sub event, topic: %v, body: %s. status code returned: %v", msg.topic, body, statusCode)
```

Comment:
Checked this one - this is done as it is done for all the left status codes, considering them as retriable. This is totally in sync with normal subscribe as well. And, the body here is of response, so that user can check back what message got passed back in response:
```	_, body := resp.RawData()
```

I don't think this should be considered as PII then, but if the view is against this, then I can remove it.

Reply:
I'd say that we should probably sanitize it out of all of them. I feel like this is the generally accept practice but maybe I'm just being over cautious. Thoughts @artursouza?

Regardless, we can do that as a separate issue once we decide on it, especially if it's the existing path.

Reply:
We should not log body since we don't know the content. It can also impact perf. At best, it can be a setting where it is OFF by default.

Reply:
Removed setting of response `body` from all the places for now.